classes/Search.php

```diff
@@ -750,7 +750,7 @@ public static function removeProductsSearchIndex($products)
     protected static function setProductsAsIndexed(&$products)
     {
         if (is_array($products) && !empty($products)) {
-            ObjectModel::updateMultishopTable('Product', array('indexed' => 1), 'a.id_product IN ('.implode(',', $products).')');
+            ObjectModel::updateMultishopTable('Product', array('indexed' => 1), 'a.id_product IN ('.implode(',', array_map('intval', $products)).')');
```

Comment:
If we use it here, we have to use everywhere to keep the same logic ?

For the sql, it's not a problem if array have duplicate value.
Moreover, `$products_array = array_unique($products_array);` is done juste before calling this function.